### ISSUE TITLE
`names_c` in calculateReliabilities()

### ISSUE THREAD

**Issue by emstruong** [FloSchuberth/cSEM#534]
Currently, `names_c` is set as

```r
names_c  <- setdiff(names(.csem_model$construct_type), names_cf)
```
whereas `names_cf` is set as

```r
names_cf <- names(.csem_model$construct_type[.csem_model$construct_type == "Common factor"])
```

Future update should consider simplifying `names_c` to

```r
names_c  <- names(.csem_model$construct_type[.csem_model$construct_type == "Composite"])
```
for readability and verify that this change doesn't break any functionality

**Comment by FloSchuberth** [comment]
Hey Michael,

What do you want to achieve by the change from common factor to composite? Do.you want to calculate the reliability for composites?
By assumption the indicators making up a composite are free from random measurement error and therefore, the reliability should be always one. Currently, we set them in case of composites to 1.

Best regards,
Flo
________________________________
From: emstruong ***@***.***>
Sent: Monday, May 20, 2024 17:47
To: FloSchuberth/cSEM ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [FloSchuberth/cSEM] `names_c` in calculateReliabilities() (Issue #534)


Currently, names_c is set as

names_c  <- setdiff(names(.csem_model$construct_type), names_cf)

whereas names_cf is set as

names_cf <- names(.csem_model$construct_type[.csem_model$construct_type == "Common factor"])

Future update should consider simplifying names_c to

names_c  <- names(.csem_model$construct_type[.csem_model$construct_type == "Composite"])

for readability and verifying whether that this doesn't break any functionality

**Comment by emstruong** [comment]
Hey Flo,

I think there's a mis-understanding--I don't think this should be changing from common factor to composite, or changing any functionality. This is just purely a coding change for stylistic purposes, the functionality should be the same. 

Please see the following screenshot

![image](https://github.com/FloSchuberth/cSEM/assets/112967876/21e88890-2141-4d4b-8c29-f898bd133f1d)

**Comment by FloSchuberth** [comment]
I see and yes, I think this should work. 
I have also created a new issue as I want to implement more test using testthat. I just need to figure out when...:/

**Comment by emstruong** [comment]
> I see and yes, I think this should work. I have also created a new issue as I want to implement more test using testthat. I just need to figure out when...:/

Well, I was thinking of cleaning all the tests up and everything before we attempt to publish the package somewhere... So perhaps after the summer-ish?

**Comment by FloSchuberth** [comment]
I just thought having more tests would facilitate detecting problems if we change sth. in the code. Particularly, adding more models, e.g., different types of second-order constructs, or single-indicator constructs. Moreover, I want to use an empirical dataset not only population data, In this way, I hope we can better figure out whether our changes in csem will have negative sideeffects. 

Yes, after the summer, I should also have a bit more time. Let's see perhaps I also find time before....

**Comment by emstruong** [comment]
> I just thought having more tests would facilitate detecting problems if we change sth. in the code. Particularly, adding more models, e.g., different types of second-order constructs, or single-indicator constructs. Moreover, I want to use an empirical dataset not only population data, In this way, I hope we can better figure out whether our changes in csem will have negative sideeffects.
> 
> Yes, after the summer, I should also have a bit more time. Let's see perhaps I also find time before....

Yes, I agree RE: using both empirical and simulated population data... I think they address different problems. Empirical for testing the user-experience and potential errors induced by user-input; versus simulated population data for true technical errors within the cSEM library itself... Perhaps we should release another version on CRAN once the new tests are complete and IGSCA Trees is completely integrated

**Comment by FloSchuberth** [comment]
In addition, using empirical data can help spotting problems that remain undetected using simulated data. 
Yes releasing a new CRAN version makes sense once we have a running (I)GSCA(m) implementation. So once, we merge your GSCA branch, we should take care of that.
